include/zephyr/lorawan/lorawan.h

```diff
@@ -318,6 +318,25 @@ void lorawan_get_payload_sizes(uint8_t *max_next_payload_size,
  */
 int lorawan_set_region(enum lorawan_region region);
 
+/**
+ * @brief Request for time according to DeviceTimeReq MAC cmd
+ *
+ * Append MAC DevTimeReq command. It will be processed on next send
+ * message or force sending empty message to request time immediately.
+ *
+ * @param force_request Force sending an empty message execute request
+ * @param sync_call Block function call until answer
+ * @return 0 if successful, negative errno otherwise
+ */
+int lorawan_request_time(bool force_request, bool sync_call);
+
+/**
+ * @brief Read current time from LoRaWAN stack
```

Comment:
Comment should probably make a note that the time returned is seconds from the GPS epoch